app/assets/javascripts/discourse/app/components/sidebar/edit-navigation-menu/categories-modal.gjs

```diff
@@ -18,126 +18,142 @@ import gt from "truth-helpers/helpers/gt";
 import includes from "truth-helpers/helpers/includes";
 import not from "truth-helpers/helpers/not";
 
+// Given a list, break into chunks starting a new chunk whenever the predicate
+// is true for an element.
+function splitWhere(elements, f) {
+  return elements.reduce((acc, el, i) => {
+    if (i === 0 || f(el)) {
+      acc.push([]);
+    }
+    acc[acc.length - 1].push(el);
+    return acc;
+  }, []);
+}
+
+function findAncestors(categories) {
```

Comment:
This is probably going to be useful in other places where we process categories, right? Worth seeing if it can be moved to the JS category model or a similar general-purpose location. (Not a blocker though.)